wallet-new/test/unit/Wallet/Inductive/Cardano.hs

```diff
@@ -223,12 +223,12 @@ equivalentT activeWallet (pk,esk) = \mkWallet w ->
                )
             => Text
             -> (Wallet h Addr -> a)
-            -> (Kernel.PassiveWallet -> IO (Interpreted a))
+            -> (Kernel.DB -> Interpreted a)
             -> TranslateT (EquivalenceViolation h) m ()
         cmp fld f g = do
           let dsl = f inductiveCtxtWallet
           translated <- toCardano ctxt fld dsl
-          kernel     <- liftIO $ g passiveWallet
+          kernel     <- g <$> liftIO (Kernel.getWalletSnapshot passiveWallet)
```

Comment:
@adinapoli-iohk `checkWalletState` calls `cmp` with multiple queries, so you can move the `getWalletSnapshot` there instead and pass the snapshot into `cmp`. That will also demonstrate the intent of this PR nicely.

Reply:
I see! The reason I didn't do it right off the bat is that I wasn't sure we needed access to the full `PassiveWallet`. Will change!